helix-term/src/commands.rs

```diff
@@ -1521,6 +1521,24 @@ mod cmd {
         }
     }
 
+    /// Sets the [`Document`]'s encoding..
+    fn set_encoding(cx: &mut compositor::Context, args: &[&str], _: PromptEvent) {
+        let (_, doc) = current!(cx.editor);
+        if let Some(label) = args.first() {
+            doc.set_encoding(label)
+                .unwrap_or_else(|e| cx.editor.set_error(e.to_string()));
+        } else {
+            let encoding = doc.encoding().name().to_string();
+            cx.editor.set_status(encoding)
+        }
+    }
+
+    /// Reload the [`Document`] from its source file.
+    fn reload(cx: &mut compositor::Context, _args: &[&str], _: PromptEvent) {
+        let (view, doc) = current!(cx.editor);
+        doc.reload(view.id).unwrap();
+    }
```

Comment:
I haven't totally thought through all the corner-cases of the workflow I was expecting yet.  This is already pretty close.

But I think what I'd probably want is that reloading the document by default still does the encoding auto-detect, rather than pulling from the in-memory document's specified encoding.  And then when I want to force-reload the document with a different encoding, I specify that as a parameter to this command.

But I'm not sure if that's *actually* what I want.  Guess it would take some playing around.  So probably leave this for now, and we can change it later if we decide to.